lib/watch.js

```diff
@@ -37,18 +37,15 @@ function assertEncoding(encoding) {
 }
 
 function guard(fn) {
-  return function(arg, action) {
-    if (is.func(fn)) {
-      var ret = fn(arg, SKIP_FLAG);
-      if (ret && ret !== SKIP_FLAG) action();
```

Comment:
There's no way to do comparison once `SKIP_FLAG` is removed  from the argument list.

```js
watch(__dirname, { 
  filter(name, flag) {
    // The flag value is always passed to this callback function so it can be returned back.
  }
});
```
If I recall correctly, there are three kinds of returning value:  `true`, `false` and `skip`(symbol).

Reply:
The flag is being passed, but it is just `false` instead of SKIP_FLAG. This means that when they return `skip`, it will just be `false`, which means ignore the item.

That way, you can check if you need to do extra tests for directory ignoring: only if `skip` is truthy.